node/src/service.rs

```diff
@@ -299,10 +257,14 @@ where
     }
 
     async fn import_block(&self, block: BlockImportParams<B>) -> Result<ImportResult, Self::Error> {
-        // Import like Frontier, but fallback to grandpa import for errors
-        match ensure_log(block.header.digest()).map_err(Error::from) {
-            Ok(()) => self.inner.import_block(block).await.map_err(Into::into),
-            _ => self.inner.import_block(block).await.map_err(Into::into),
+        // 4345556 - mainnet runtime upgrade block with Frontier
+        if *block.header.number() < 4345557u32.into() {
```

Comment:
This matches the comment a little better:
```suggestion
        if *block.header.number() <= 4345556u32.into() {
```

Flipping if/else after that would be even more clear.

If someone were to search through the codebase, they are more likely to be searching for `4345556` rather than `4345557` too.